client/src/components/Form/Elements/FormCheck.vue

```diff
@@ -25,26 +23,17 @@ const currentValue = computed({
         } else {
             emit("input", null);
         }
-        if (newValue.length === 0) {
-            selectAll.value = false;
-            indeterminate.value = false;
-        } else if (newValue.length === props.options.length) {
-            selectAll.value = true;
-            indeterminate.value = false;
-        } else {
-            indeterminate.value = true;
-        }
     },
 });
 
-const hasOptions = computed(() => {
-    return props.options.length > 0;
-});
-
-const selectAll = ref(false);
+const hasOptions: ComputedRef<boolean> = computed(() => props.options.length > 0);
+const indeterminate: ComputedRef<boolean> = computed(
+    () => ![0, props.options.length].includes(currentValue.value.length)
+);
+const selectAll: ComputedRef<boolean> = computed(() => currentValue.value.length === props.options.length);
 
-function onSelectAll() {
-    if (selectAll.value) {
+function onSelectAll(selected) {
```

Comment:
```suggestion
function onSelectAll(selected: boolean) {
```
that's the only one that can't be inferred